src/nspawn/nspawn.c

```diff
@@ -2589,13 +2595,45 @@ static int setup_seccomp(void) {
                 goto finish;
         }
 
-        for (i = 0; i < ELEMENTSOF(blacklist); i++) {
-                r = seccomp_rule_add(seccomp, SCMP_ACT_ERRNO(EPERM), blacklist[i], 0);
-                if (r == -EFAULT)
-                        continue; /* unknown syscall */
-                if (r < 0) {
-                        log_error_errno(r, "Failed to block syscall: %m");
-                        goto finish;
+        /* If the CAP_SYS_RAWIO capability is not requested,
```

Comment:
Regardless of how the rest of this file appears, it would be nice to not repeat yourself 3 times here. Can't you abstract this into one function? It appears the only difference between the three copies are constants.